### ISSUE TITLE
No solution is returned for `solve(exp(log(5)*x) - exp(log(2)*x), x)`

### ISSUE THREAD

**Issue by sidhantnagpal** [sympy/sympy#14791]
Transcendental Equation:
```
In [1]: solve(exp(log(5)*x) - 2**x, x)
Out[1]: [0]

In [2]: solve(exp(log(5)*x) - exp(log(2)*x), x)
Out[2]: []
```

Original Comment: PR [#14712](https://github.com/sympy/sympy/pull/14712#issuecomment-393174502)

**Comment by smichr** [comment]
```python
>>> var('x',real=True)
x
>>> solve(exp(log(5)*x) - exp(log(2)*x), x)
[0]
```

**Comment by sidhantnagpal** [comment]
Fixed in #14712
